### PR TITLE
WIP: Simplifies eslint config

### PR DESCRIPTION
**Pull request by zomars**
I've noticed we're overcomplicating our eslint config. I'm trying to simplify this by falling back to the Next's eslint config as much as possible.

### CODE REVIEW

**Comment by emrysal** [comment]
I'm not fully aware of all the things happening within the Eslint config; but many of the explicit omissions have been put there for a purpose so I'm not sure I agree with everything in here. @KATT knows more about what's going on; though; so I'll defer to him for providing more background info.

File: .lintstagedrc.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  "**/*.(j|t)s?(x)": (filenames) => [
+    `tsc-files --noEmit ${filenames.map((file) => "." + file.split(process.cwd())[1]).join(" ")}`,
+    `next lint --fix --file ${filenames.map((file) => file.split(process.cwd())[1]).join(" --file ")}`,
+  ],
+  "./prisma/schema.prisma": "prisma format",
+};
```
Comment (KATT) [inline .lintstagedrc.js:7]: why is it better to have this here?

File: .eslintrc.json [hunk 2]
```diff
@@ -1,48 +1,3 @@
...
-    "node": true,
-    "es6": true,
-    "jest": true
-  },
-  "settings": {
-    "react": {
-      "version": "detect"
-    }
-  }
+  "extends": ["next", "next/core-web-vitals", "prettier"]
```
Comment (KATT) [inline .eslintrc.json:2]: Are you sure non-standard folders are linted with Next.js standard linting? i.e. `/server`, `lib` -- I have a faint memory of encountering something like that before

**Review by KATT — APPROVED** [review]
LGTM, just double-check linting runs on all folders and not only like `pages/`

**Comment by zomars** [comment]
It's still WIP. Doing some more testing @KATT

File: .lintstagedrc.js (hunk 1 above)
Comment (zomars) [reply .lintstagedrc.js:7]: Next lint was having some issues with the package.json format

File: .eslintrc.json (hunk 2 above)
Comment (zomars) [reply .eslintrc.json:2]: Will double check

**Comment by zomars** [comment]
This will be handled better in our mono repo

**Pull request closed without merging**